### PR TITLE
Noetic release

### PR DESCRIPTION
**Pull request by ahcorde**
Following the guidelines to migrate packages to [noetic](http://wiki.ros.org/noetic/Migration)
 - Python3 changes

 - import setup from setuptools instead of distutils-core

 - Bump CMake version to avoid CMP0048 warning
This bumps the minimum CMake version to 3.0.2, which is the [minimum supported by ROS Kinetic](https://www.ros.org/reps/rep-0003.html#kinetic-kame-may-2016-may-2021) and new enough to default to the NEW behavior of [CMP0048](https://cmake.org/cmake/help/v3.0/policy/CMP0048.html). This avoids a CMake warning when building and testing this package in Debian Buster and Ubuntu Focal.

### CODE REVIEW

**Comment by v4hn** [comment]
@ahcorde Looks like your python3 change (for some reason?) made the test fail. Can you have a look?

**Comment by ahcorde** [comment]
Noetic should work on Ubuntu Focal and Python3. In the error I see that tests are being launched with Python2.7

```
<testsuite errors="1" failures="0" name="unittest.suite.TestSuite" tests="3" time="0.046"><testcase classname="__main__.TestSRDFParser" name="test_complex_srdf" time="0.0229" /><testcase classname="__main__.TestSRDFParser" name="test_full_srdf" time="0.0024"><error type="AttributeError">'unicode' object has no attribute 'nodeType'
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/root/ros_ws/src/srdfdom/test/test.py", line 147, in test_full_srdf
    self.assertTrue(xml_matches(robot.to_xml_string(),expected))
  File "/root/ros_ws/src/srdfdom/test/test.py", line 75, in xml_matches
    return xml_matches(a, parseString(b).documentElement)
  File "/root/ros_ws/src/srdfdom/test/test.py", line 76, in xml_matches
    if a.nodeType == xml.dom.Node.DOCUMENT_NODE:
```

**Comment by v4hn** [comment]
> In the error I see that tests are being launched with Python2.7

I assume you want to say your patch is not python2/3 compatible?
In that case I just pushed a new branch `noetic-devel` and retargeted your request.

**Comment by ahcorde** [comment]
I have created a [source job](https://github.com/ros/rosdistro/pull/24326) to be able to test it in the buildfarm in the right conditions

**Comment by v4hn** [comment]
Thanks, MoveIt's [own CI is currently not yet up to the task](https://github.com/ros-planning/moveit_ci/issues/108).

I guess we want to get our CI ready for it before merging this though.

**Comment by rhaschke** [comment]
There is not even a ROS Noetic docker image available: https://hub.docker.com/_/ros?tab=tags&page=1&name=noetic
So, we cannot yet setup our CI. I asked @sloretz at last ROSCon already to provide this infrastructure.

**Comment by ahcorde** [comment]
I'm working in to unblock some metapackages to be able to create the Docker images.

**Comment by rhaschke** [comment]
Is there a list of pending packages somewhere?

**Comment by ahcorde** [comment]
http://repositories.ros.org/status_page/blocked_releases_noetic.html?s=6&r=1

**Comment by v4hn** [comment]
That's quite a list... wow. Thank you for working on it.

Is there also a list of all packages required for the docker image?

**Comment by ahcorde** [comment]
Here you can find the specific list for the metapackages repository: http://repositories.ros.org/status_page/blocked_releases_noetic.html?q=id%3D%22metapackages%22&s=6&r=1

**Review by rhaschke — APPROVED** [review]

**Comment by v4hn** [comment]
@rhaschke you just merged this in the new `noetic-devel` branch.
I agree that it would be much better to have the `melodic-devel` branch python3 compatible!

Feel free to merge it back there and drop the noetic branch again for now.

**Comment by rhaschke** [comment]
For now, melodic-devel and noetic-devel are on par.

**Comment by sloretz** [comment]
> There is not even a ROS Noetic docker image available: https://hub.docker.com/_/ros?tab=tags&page=1&name=noetic
So, we cannot yet setup our CI. I asked @sloretz at last ROSCon already to provide this infrastructure.

@ahcorde added a source entry for `srdfdom` enabling pull request jobs for this repo in https://github.com/ros/rosdistro/pull/24326 . Assuming [this repo is setup for PR testing on the ROS buildfarm](http://wiki.ros.org/buildfarm/Pull%20request%20testing), PRs will be tested by [this PR job](http://build.ros.org/job/Npr__srdfdom__ubuntu_focal_amd64/), and the `noetic-devel` branch will be tested by [this dev job](http://build.ros.org/job/Ndev__srdfdom__ubuntu_focal_amd64/).

**Pull request merged**
